### PR TITLE
Update deps to bring in support for OpenBSD

### PR DESCRIPTION
**Pull request by qbit**
#### Name of feature:

Let `step-ca` build on OpenBSD.

With these deps bumped, tests all pass on OpenBSD amd64.

#### Supporting links/other PRs/issues:

OpenBSD support was added to the following deps:

 - github.com/go-piv/piv-go in https://github.com/go-piv/piv-go/pull/101
 - github.com/newrelic/go-agent in https://github.com/newrelic/go-agent/pull/455
 - github.com/miekg/pkcs11 in https://github.com/miekg/pkcs11/pull/140

### CODE REVIEW

**Comment by tashian** [comment]
Thanks for putting this together @qbit!

File: go.mod [hunk 1]
```diff
@@ -34,8 +34,8 @@ require (
 	github.com/mattn/go-colorable v0.1.8 // indirect
 	github.com/mattn/go-isatty v0.0.13 // indirect
 	github.com/micromdm/scep/v2 v2.1.0
-	github.com/miekg/pkcs11 v1.0.3 // indirect
-	github.com/newrelic/go-agent v2.15.0+incompatible
+	github.com/miekg/pkcs11 v1.1.1 // indirect
+	github.com/newrelic/go-agent v2.16.4-0.20220510172711-aee135dc5dcf+incompatible
```
Comment (maraino) [inline go.mod:38]: It looks like the commit https://github.com/newrelic/go-agent/commit/aee135dc5dcf93df9dec7d55ab6ae6d40c0a4d7b is tagged with `v3.18.0`, can you use that instead?

File: go.mod [hunk 2]
```diff
@@ -20,7 +20,7 @@ require (
 	github.com/form3tech-oss/jwt-go v3.2.3+incompatible // indirect
 	github.com/go-chi/chi v4.0.2+incompatible
 	github.com/go-kit/kit v0.10.0 // indirect
-	github.com/go-piv/piv-go v1.7.0
+	github.com/go-piv/piv-go v1.9.1-0.20220211232127-7e03e7f49e54
```
Comment (maraino) [inline go.mod:23]: I've noticed you created the PR for supporting OpenBSD on `piv-go`. I will accept this version, but can you request `piv-go` maintainers to create a new release?

**Review by maraino — CHANGES_REQUESTED** [review]
@qbit Thanks for this, but can you change the version of the go-agent and if possible get a new release of go-piv.

File: go.mod (hunk 1 above)
Comment (qbit) [reply go.mod]: https://github.com/go-piv/piv-go/issues/105 Done.

File: go.mod (hunk 2 above)
Comment (qbit) [reply go.mod:23]: Done: https://github.com/go-piv/piv-go/issues/105

**Comment by qbit** [comment]
OK, updated go-agent and made a PR to go-piv for cutting a new version.

**Comment by qbit** [comment]
yay! go-piv has a new release, I'll update to include that.

**Comment by qbit** [comment]
Ok, both updates are pointing to tagged releases!

**Review by maraino — APPROVED** [review]
Thanks @qbit.

I'll update https://github.com/smallstep/step-kms-plugin with that version of piv-go and pkcs11. Check it out, is a plugin to `step` to use `step certificate create` and `step certificate sign` with PKCS#11 modules, yubikeys, CloudKMS, ...

**Comment by maraino** [comment]
@qbit I'm gonna take a look at what's going on with the linker before merging ...

**Pull request merged**